### PR TITLE
Allow users to specify the topology key for the anti-affinity check

### PR DESCRIPTION
**Pull request by viswajithiii**
This allows users to allow inter-pod anti-affinity to use an arbitrary label (example: `topologyKey: failure-domain.beta.kubernetes.io/zone`), not just `kubernetes.io/hostname`.

### CODE REVIEW

**Review by kreamkorokke — APPROVED** [review]
Now, thought about adding a unit test?;)

**Comment by kreamkorokke** [comment]
And it's fine if you got some other stuffs on hand, we can add them gradually over time as well

File: internal/templates/templates_testutils.go [hunk 1]
```diff
@@ -35,15 +37,18 @@ func (s *TemplateTestSuite) Validate(
...
-			s.Error(err, "param should have caused error but did not raise one")
-			continue
-		}
-		for _, obj := range ctx.Objects() {
-			diagnostics := checkFunc(ctx, obj)
-			s.compareDiagnostics(c.Diagnostics, diagnostics)
-		}
+		s.Run(fmt.Sprintf("%+v", c.Param), func() {
+			checkFunc, err := s.Template.Instantiate(c.Param)
+			if c.ExpectInstantiationError {
```
Comment (kreamkorokke) [inline internal/templates/templates_testutils.go:42]: should this be `c.ExpectInstantiationError && err == nil`?

**Review by kreamkorokke — APPROVED** [review]

File: internal/templates/templates_testutils.go (hunk 1 above)
Comment (viswajithiii) [reply internal/templates/templates_testutils.go:42]: `s.Error(err)` inside basically fails if `err != nil` so it's correct.

**Pull request merged**